### ISSUE TITLE
Inconsistent behavior of NoReturn and try except branch

### ISSUE THREAD

**Issue by tamuhey** [microsoft/pyright#672]
Pyright incorrectly report an error for the following code:

```python
class E:
    def __call__(self):
        raise ValueError


def f():
    try:
        import pathlib
    except ImportError:
        E()()

    a = pathlib.Path() # error: "pathlib" is possibly unbound 
```

However, the next code is passed:

```
# now error is emitted from function, not class.
def e():
    raise ValueError


def g():
    try:
        import pathlib
    except ImportError:
        e()

    a = pathlib.Path() # this line passes the check
```

**Comment by erictraut** [comment]
Thanks for the issue report.

This is actually "as designed" behavior. The code that infers "NoReturn" return results needs to be fast and not rely on full type analysis. As such, it can't handle all cases, such as callable objects that always raise exceptions in their `__call__` method. You'll need to find a workaround for this in your code.

**Comment by erictraut** [comment]
I've added more logic to handle this particular case, so this will be addressed in the next release of pyright.

**Comment by erictraut** [comment]
This is addressed in pyright 1.1.199, which I just published. It will also be included in the next release of pylance.
